enhancements/etcd/dr-leveraging-ceo-scaling.md

```diff
@@ -0,0 +1,151 @@
+---
+title: cluster-etcd-operator
+authors:
+  - "@alaypatel07"
+reviewers:
+  - "@hexfusion"
+  - "@deads2k"
+  - "@retroflexer"
+approvers:
+ - TBD
+creation-date: 2020-2-21
+last-updated: 2020-2-21
+status: brainstorming
+see-also: 
+replaces:
+superseded-by:
+---
+
+# Disaster Recovery options leveraging Cluster Etcd Operator scaling up
+Currently the Disaster Recovery scripts support 3 scenarios:
+1. Recovering from total loss of control plane with ssh access to atleast 1 master node and backup
+2. Restoring the cluster state from a backup. Here the control plane is available but the etcd state is reset to that of the backup.
+3. Recovering the loss of 1 master node. This include a scenario where the maste node is up but etcd data directory is corrupted. 
+
+This document discusses the ideas involved in making the first two scenarios easiest to implement and consumed by the user. The ideas is to have  1 etcd running so control plane is up and CEO scales the remaining etcd members.  
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+TODO
+
+## Summary
+
+cluster-etcd-operator (CEO) is an operator that handles the scaling of etcd during cluster
+ bootstrap and regular operation and provisioning etcd dependencies such as TLS certificates.
+
+## Motivation
+
+TODO
+
+
+### Goals
+
+1. Leverage the scale up mechanism of CEO
+
+### Non-Goals
+
+TODO
+
+## Challenges and possible alternatives
+
+A restore operation is a two step process on each node. 
+1. Restoring the data directory with appropriate environment variables with the following operation:
+```
+ETCDCTL_API=3 etcdctl snapshot restore ./path/to/backup.db --name $ETCD_NAME --initial-cluster="$ETCD_NAME=$ETCD_NODE_PEER_URL" --initial-cluster-token openshift-etcd --initial-advertise-peer-urls $ETCD_NODE_PEER_UR  --data-dir="/path/to/new/data-dir"
+```
+2. Starting the etcd pod should be started with the same value of $ETCD_NAME, ETCD_INITIAL_CLUSTER and $ETCD_NODE_PEER_URL as in step 1.
+
+This will allow etcd to populate the data directory with a new cluster ID, and new membership data that includes only 1 member.
```

Comment:
So in you POC, we have a new clusterid?

Reply:
I did another restore to confirm, http://pastebin.test.redhat.com/839310 see first and third, they are with unique `--initial-cluster-token` value